client/my-sites/stats/stats-site-overview/index.jsx

```diff
@@ -81,3 +76,20 @@ export default React.createClass( {
 		);
 	}
 } );
+
+export default connect( ( state, ownProps ) => {
+	const { siteId, date, period, siteSlug } = ownProps;
+	const query = {
+		date,
+		period
+	};
+	// It seems not all sites are in the sites/items subtree consistently
+	const slug = siteSlug || getSiteSlug( state, siteId );
```

Comment:
Tested this on the branch and it works just fine. Do we expect to pass through a siteSlug to this component? Otherwise I would probably simplify this to:
```
const slug = getSiteSlug( state, siteId );
```